BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs

```diff
@@ -492,6 +502,30 @@ private static void SetTelemetryChannelEndpoint(ITelemetryChannel channel, strin
             }
         }
 
+        private static void SetTelemetryChannelCredentialEnvelope(ITelemetryChannel telemetryChannel, CredentialEnvelope credentialEnvelope)
+        {
+            if (telemetryChannel is InMemoryChannel inMemoryChannel)
```

Comment:
```suggestion
            if (telemetryChannel is ISupportCredentialEnvelope iSupportCredentialEnvelope)
```

Hopefully in next round.

Reply:
Yes. I removed it for this PR, but i need that interface to interact with ServerTelemetryChannel :)